chasm/ref_test.go

```diff
@@ -0,0 +1,83 @@
+package chasm
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"go.temporal.io/server/common/testing/testvars"
+	gomock "go.uber.org/mock/gomock"
+)
+
+type componentRefSuite struct {
+	suite.Suite
+	*require.Assertions
+
+	controller *gomock.Controller
+
+	registry *Registry
+}
+
+func TestComponentRefSuite(t *testing.T) {
+	suite.Run(t, new(componentRefSuite))
+}
+
+func (s *componentRefSuite) SetupTest() {
+	s.initAssertions()
+
+	s.controller = gomock.NewController(s.T())
+
+	s.registry = NewRegistry()
+	s.registry.Register(newTestLibrary(s.controller))
+}
+
+func (s *componentRefSuite) SetupSubTest() {
+	s.initAssertions()
+}
+
+func (s *componentRefSuite) initAssertions() {
+	// `s.Assertions` (as well as other test helpers which depends on `s.T()`) must be initialized on
+	// both test and subtest levels (but not suite level, where `s.T()` is `nil`).
+	//
+	// If these helpers are not reinitialized on subtest level, any failed `assert` in
+	// subtest will fail the entire test (not subtest) immediately without running other subtests.
+
+	s.Assertions = require.New(s.T())
+}
```

Comment:
imo just simplify to 
```suggestion
func (s *componentRefSuite) SetupSubTest() {
	s.Assertions = require.New(s.T())
}
```

Reply:
I removed SetupSubTest since there's no sub test in this suite right now. Also added a comment in SetupTest().